[ADD] rental: service products page (#15560)
odoo/documentation: content/applications/sales/rental/service_products.rst

================
Service products
================

The **Rental** app is a comprehensive tool that enables users to manage the scheduling, pricing, and
inventory for both physical goods (products) and non-physical goods (services) within a single
platform.

This flexibility allows for combining products and services like bike rentals with guided tours, or
booking a studio with a photographer.

> This document will cover how to configure a rental service to automatically sync with staff shifts,

Reviewer: ```suggestion
This document covers how to configure a rental service to automatically sync with staff shifts,
```

track timesheet hours, and set up and link project tasks based on a rental order.

Settings
========

> To configure default settings on rental products, navigate to :menuselection:`Reantal app -->

Reviewer: ```suggestion
To configure default settings on rental products, navigate to :menuselection:`Rental app -->
```

   :guilabel:`Per Day` fields at the product level. If the defaults apply to all products, leave the
   :guilabel:`Product` field blank.

In the :guilabel:`Default Padding Time` section, fill in the :guilabel:`Padding` field.

Next, enable :guilabel:`Rental Transfers`. In the :guilabel:`Rent Online` section, fill in the
:guilabel:`Minimal Rental Duration` field and designate :guilabel:`Unavailability days`. Click
:guilabel:`Save` to apply the changes.

App integration configuration
=============================

To allow for workflow efficiency and automation when creating a service product and rental order,
the following Odoo apps and configuration are reccomended.

Project app settings
--------------------

Install the **Project** app and enable :guilabel:`Project stages` and :guilabel:`Timesheets`. Refer
to the :doc:`Project management <../../services/planning>` page for more information.

Planning app settings
---------------------

Install the **Planning** app and configure employee shifts and roles. Refer to the :doc:`Project
management <../../services/planning>` for instructions on setting up your employees and shifts.

Sales app settings
------------------

> Install the **Sales** app. Navigate to :menuselection:`Sales app  --> Configuration --> Settings`.

Reviewer: The sales app is required for rental, right? (if not disregard). I would add a note that it is required, and move it above the other two

a price` to enter a new rental rate. Choose a *pricing period* (:dfn:`the unit of duration of the
rental`) in the Period column, or create a new pricing period by typing in the name and clicking
Create and edit.

.. tip::
   Customize rental rate time periods by navigating to :menuselection:`Rental app --> Configuration
   --> Rental periods`.

   .. image:: service_products/new-rental-period.png
      :alt: Sample of a Create Period pop up window in the Rental app.

Next, enter the :guilabel:`Price` for that specific :guilabel:`Period`. To apply the configured
rental rate to an existing pricelist, click in the :guilabel:`Pricelist` column and select the
desired list from the drop-down menu.

In the :guilabel:`Reservations` section, fill in the :guilabel:`Hourly Fine`, :guilabel:`Daily
Fine`, and the :guilabel:`Reserve product` time. These values are automatically populated from the
:guilabel:`Default Delay Costs` section, provided they have been configured in the
::menuselection:`Rental app --> Configuration --> Settings`.

Click the :icon:`fa-cloud-upload` :guilabel:`(Save manually)` icon near the top to save.

.. example::

>    A photography studio rents out their photographers on a hourly and daily basis. They want to add

Reviewer: This is a brilliant example :)

>    new four hour package for $750.

Reviewer: ```suggestion
  a new four hour package for $750.
```

>    All reservations require 24 hours notice to reserve a photographer, but they do not charge a

Reviewer: ```suggestion
   All reservations require 24-hour notice to reserve a photographer, but they do not charge a
```

   fines if the reservations go over the agreed time. Instead they default to their hourly fee.

   Create a new pricing period by navigating to :menuselection:`Rental app --> Configuration -->

>    Rental periods`. Click :guilabel:`New` and configure a the period for four hours.

Reviewer: ```suggestion
   Rental periods`. Click :guilabel:`New` and configure the period for four hours.
```

   Navigate to the :guilabel:`Photographer` service product and in the :guilabel:`Rental prices`
   tab, add the four hour period set at $750. Manually save to apply changes.

   .. image:: service_products/example-4hrs-rental-period.png
      :alt: Sample of the Rental prices tab of service product in the Rental app.

Create a rental order with a service product
============================================

Naviagte to the **Rental** app which to opens the Rental Orders dashboard. Click :guilabel:`New` to
open a new rental order form.

Enter the :guilabel:`Customer` field, and select a :guilabel:`Quotation Template`, if applicable.

Next, set the desired rental duration in the :guilabel:`Rental period` field. To adjust the rental
duration, click the first date in the :guilabel:`Rental period` field, and select the range of dates
and times to represent the rental duration from the pop-up calendar form that appears.

.. image:: service_products/rental-period-field-popup.png
   :alt: Sample of the rental period field calendar pop-up window in the Rental app.

Once complete, click :guilabel:`Apply` in the calendar pop-up form. Following that, the pop-up form

> disappears, and the designated time period of the rental is represented in the :guilabel:`Duration`

Reviewer: ```suggestion
disappears, and the designated rental time period is represented in the :guilabel:`Duration`
```

   represent the duration of the rental, then click Update Rental Prices in the Duration field.

   .. image:: service_products/rental-update-rental-prices.png
      :alt: Sample of the Duration field with the Update Rental Prices option in the Rental app.

   Doing so reveals a Confirmation pop-up window. If everything is correct, click Ok, and Odoo
   recalculates the rental price accordingly.

.. image:: service_products/rental-order-form.png
   :alt: Sample of a rental order for a service product in the Rental app.

Once all information has been entered correctly on the rental order form, click :guilabel:`Send` to
email the quotation to the customer. When the customer approves the quotation, click
:guilabel:`Confirm`. A banner displays on the rental order stating its current status.

Customer signature
==================

Odoo allows you to request that the customer sign a rental agreement using an electronic signature.
Rental agreements outline the arrangement between the company and the customer before they pick up
the rental products. Such documents can ensure everything is returned on-time and in its original

> condition.

Reviewer: ```suggestion
the rental products. 
```
Again, since this isn't a normal product rental, I don't know that this fits. Maybe something about service expectations?

.. note::
   Requesting an electronic signature can be done during any stage of the order. This feature also
   requires the **Sign** app.

If signatures are required, go to the **Rental** app and from the default :guilabel:`Rental Orders`

> dashboard, select the desired rental order. To go the :icon:`fa-cog` :guilabel:`(Actions)` icon, and

Reviewer: ```suggestion
dashboard, select the desired rental order. Click the :icon:`fa-cog` :guilabel:`(Actions)` icon, and
```

.. image:: service_products/request-signature.png
   :alt: The Action drop-down menu for requesting a signature in the Rental app.

A :guilabel:`New Signature Request` pop-up window displays. Select the desired document from the
:guilabel:`Template` drop-down menu.

.. image:: service_products/new-signature-request.png
   :alt: Sample of the New Signature Request pop-up window in the Rental app.

Doing so reveals a :guilabel:`New Signature Request` pop-up window. Upon confirming the information
in the :guilabel:`New Signature Request` pop-up form, click :guilabel:`Send` to initiate the signing
process.

.. image:: service_products/new-signature-request-message.png
   :alt: Sample of the New Signature Request message in the Rental app.

A link to the signature request will appear in the chatter of the rental order. The document is
accessible to the customer via the customer portal or email.

Entering time for the rental order
==================================

.. important::
   For the appropriate smart buttons to display, the **Project**, **Timesheet**, and **Planning**
   apps are needed. The selected service product on the rental order must be :ref:`properly

>    configured <rental/service_products/service>` to integrate with the reccomended apps.

Reviewer: ```suggestion
   configured <rental/service_products/service>` to integrate with the recommended apps.
```

invoice to the customer or :guilabel:`Print` and then click :guilabel:`Pay` if the customer is in
person.

In the :guilabel:`Pay` pop-up window, select a :guilabel:`Journal` and click :guilabel:`Create
Payment`.

.. image:: service_products/pay-popup.png
   :alt: Sample of the Pay pop-up window in the Rental app.

Click on the :guilabel:`Payments` smart button that appears on the top of the rental order. Click
:guilabel:`Validate` on the Payment page.

.. image:: service_products/payment-validate.png
   :alt: Sample of the Payment page in the Rental app.

Use the breadcrumbs to return to the rental order.

Signing a document from a email link
------------------------------------

When a customer clicks :guilabel:`Sign document`, a separate page is then revealed, showcasing the
document to be signed. The customer begins the process by clicking :guilabel:`Click` to start.

The app guides the signee to the required signature locations and allows them to create electronic
signatures to complete the form.

.. image:: service_products/adopt-signature-popup.png
   :alt: Sample of a customer signature with the Auto option chosen.

> The adopt your signature pop-up window that appears in the Odoo Rental application. Once the

Reviewer: ```suggestion
The *Adopt your signature* pop-up window that appears in the Odoo Rental application. Once the
```

=======================
Service rental products
=======================

The **Rental** app is a comprehensive tool that enables users to manage the scheduling, pricing, and
inventory for both physical goods (products) and non-physical goods (services) within a single
platform.

This flexibility allows for combining products and services like bike rentals with guided tours, or
booking a studio with a photographer.

This document covers how to configure a rental service to automatically sync with staff shifts,
track timesheet hours, and set up and link project tasks based on a rental order.

Settings
========

To configure default settings on rental products, navigate to :menuselection:`Rental app -->
Configuration --> Settings`.

.. image:: service_products/rental-settings.png

>    :alt: Sammple of Rental app configuration settings.

Reviewer 2: ```suggestion
   :alt: Sample of Rental app configuration settings.
```

This typo isn't going to be seen by most people but since I caught, let's change it 😂

The following Odoo apps and configurations are required. The listed apps allow for workflow
efficiency and automation when creating a service product and rental order.

Sales app settings
------------------

Install the **Sales** app. Navigate to :menuselection:`Sales app  --> Configuration --> Settings`.
In the :guilabel:`Quotations & Orders` section, tick the :guilabel:`Online Signature`,
:guilabel:`Online Payment`, and :guilabel:`Quotation Templates` checkboxes.

In the :guilabel:`Default Template` field, click the :guilabel:`Quotation templates` to customize
the quotation templates to use.

Project app settings
--------------------

Install the **Project** app and enable :guilabel:`Project stages` and :guilabel:`Timesheets`. Refer
to the :doc:`Planning <../../services/planning>` page for more information.

Planning app settings
---------------------

Install the **Planning** app and configure employee shifts and roles. Refer to the :doc:`Project
management <../../services/planning>` for instructions on setting up your employees and shifts.

Sign app settings
-----------------

Install the **Sign** app. Refer to the :doc:`Sign <../../productivity/sign>` page for instructions

> on uploading a rental agreement for the business to use.

Reviewer 2: Wondering if it's overkill to explain why each app is used. A quick sentence explaining "the Projects app is with the Planning app to automatically coordinate purchased services and assign them to employees with availability" (feel free to reword this!!)

Author: I will do this next PR.

Rental services
===============

To view all products that can be rented in the database, navigate to :menuselection:`Rentals app -->
Products`. By default, the :guilabel:`Rental` filter appears in the search bar, and the view is
Kanban. Remove the filter, then click the search bar. From the preset filters, select
:guilabel:`Services`. All the configured services appear.

Each Kanban card displays the name and rental price of the service.

.. _rental/service_products/service:

Create a new service product

> ============================

Reviewer 2: Nitpick and totally optional, but I'm wondering if we should make this a h3 heading instead of h2. Since this is a step in configuring a rental service

```suggestion
----------------------------
```

(I doubt I suggested the correct number of dashes)

To set up a new rental service, go to the :menuselection:`Rental app --> Products` and then click
:guilabel:`New`. In the new product window, the :guilabel:`Rental` checkbox is already ticked by
default.

.. image:: service_products/service-product.png
   :alt: Sample of a configured service product in the Rental app.

Tick the :guilabel:`Sales` checkbox to enable the :guilabel:`Create on Order` and

> :guilabel:`Invoicing Policy` fields. Select the :guilabel:`Product Type` as a :guilabel:`Service`.

Reviewer 2: I think we can reorder this info here.

I'm creating a rental product from scratch and pls pretend im a dumbdumb new user.

On the new product form, the Product Type is set to Goods. I followed your instructions and ticked the Sales checkbox. 

I got confused and couldn't find the Create on Order and Invoicing Policy fields. I got stuck ):

Then I skipped to the next instruction and set the Product Type to service and saw the fields I couldn't find earlier.

I'm thinking we remove this part altogether so people following your instructions don't try to look for these fields. If you think it's important for the users to know that ticking Sales makes those fields appear, consider putting that info in an admonition block so the user doesn't treat it as an instruction

```suggestion
Tick the :guilabel:`Sales` checkbox. Select the :guilabel:`Product Type` as a :guilabel:`Service`.
```

   .. image:: service_products/rental-update-rental-prices.png
      :alt: Sample of the Duration field with the Update Rental Prices option in the Rental app.

   Doing so reveals a Confirmation pop-up window. If everything is correct, click Ok, and Odoo
   recalculates the rental price accordingly.

.. image:: service_products/rental-order-form.png
   :alt: Sample of a rental order for a service product in the Rental app.

Once all information has been entered correctly on the rental order form, click :guilabel:`Send` to
email the quotation to the customer. When the customer approves the quotation, click
:guilabel:`Confirm`. A banner displays on the rental order stating its current status.

Customer signature
==================

Odoo enables electronic signature requests for customer service agreements and any other documents
that require signatures. Service agreements detail the business relationship and mutual duties.
These agreements protect both the provider and the customer by creating clear, enforceable
guidelines.

.. note::
   Requesting an electronic signature can be done at any stage of the order. This feature also

>    requires the **Sign** app.

Reviewer 2: As discussed in person, in a follow-up PR, let's add more context here explaining that this getting customer signature process is unique to the services products and sometimes we *have* to get the signature at this stage.

If signatures are required, go to the **Rental** app and from the default :guilabel:`Rental Orders`
dashboard, select the desired rental order. Click the :icon:`fa-cog` :guilabel:`(Actions)` icon, and
click :guilabel:`Request Signature`.

.. image:: service_products/request-signature.png
   :alt: The Action drop-down menu for requesting a signature in the Rental app.

A :guilabel:`New Signature Request` pop-up window displays. Select the desired document from the
:guilabel:`Template` drop-down menu.

.. image:: service_products/new-signature-request.png
   :alt: Sample of the New Signature Request pop-up window in the Rental app.

Doing so reveals a :guilabel:`New Signature Request` pop-up window. Upon confirming the information
in the :guilabel:`New Signature Request` pop-up form, click :guilabel:`Send` to initiate the signing
process.

.. image:: service_products/new-signature-request-message.png
   :alt: Sample of the New Signature Request message in the Rental app.

A link to the signature request will appear in the chatter of the rental order. The document is
accessible to the customer via the customer portal or email.

Signing a document from an email link

> ------------------------------------

Reviewer 2: ```suggestion
-------------------------------------
```
missing hypen that's causign the automatic checks to fail

bottom of the document.

Odoo presents the option to download the signed document for record-keeping purposes, if necessary.

.. image:: service_products/validate-send-doc-button.png
   :alt: Sample of a signed document with the Validate & Send Completed Document button.

.. seealso::
   `Odoo Tutorials: Sign <https://www.odoo.com/slides/sign-61>`_.

Entering time for the rental order
==================================

.. important::
   For the appropriate smart buttons to display, the **Project**, **Timesheet**, and **Planning**
   apps are needed. The selected service product on the rental order must be :ref:`properly
   configured <rental/service_products/service>` to integrate with the recommended apps.

Navigate to the rental orders by :menuselection:`Rental app --> Orders --> Orders` and select the
desired rental order. At the top of the rental order form, the following smart buttons appear:

- **Tasks**: Linked to the Projects app and shows any projects or tasks related to the rental order.

- **Recorded**: Linked to the Timesheets app and shows how many hours are related to the rental
  order.

> - **Planned**: Linked to the Planning app and shows how many shifts are related to the rental order.

Reviewer 2: we bold app names !

Reviewer 2: Also the blank spaces aren't needed
```suggestion
- **Tasks**: Linked to the Projects app and shows any projects or tasks related to the rental order.
- **Recorded**: Linked to the Timesheets app and shows how many hours are related to the rental
  order.
- **Planned**: Linked to the Planning app and shows how many shifts are related to the rental order.
```

Click the :guilabel:`Tasks` smart button to view a Kanban view of all the associated tasks that were
automatically created when confirming the rental order. Click the desired task, then select the
:guilabel:`Timesheets` tab.

.. image:: service_products/rental-order-task-card.png
   :alt: Sample of a project task card for the rental order in the Rental app.

Click :guilabel:`Add a line` to enter the number of hours worked on the task manually. Click on the
:guilabel:`Sales Order` smart button to return to the rental order.

.. image:: service_products/rental-task-timesheets-tab.png
   :alt: Sample of a a task timesheet tab for a rental order in the Rental app.

.. note::
   Once time is added to the :guilabel:`Timesheets` tab  of a task, the status of the rental order
   is automatically changed to :guilabel:`Picked-up` and the :guilabel:`Return` button appears.

Create an invoice
=================

> Navigate to the desired invoice by navigating to the **Rental** app and on the :guilabel:`Rental

Reviewer 2: since you're asking someone to go or navigate, let's use menuselection even if it's just the app name

```suggestion
Navigate to the desired invoice by navigating to the :menuselection:`Rental` app and on the :guilabel:`Rental
```
